app/assets/src/components/visualizations/heatmap/Heatmap.js

```diff
@@ -530,36 +530,40 @@ export default class Heatmap {
     // If there's a specific row we want to focus on, auto-scroll the heatmap
     // so the row is centered on screen.
     const row = this.rowLabels.filter(rowLabel => rowLabel.label === label)[0];
-    const rowIndex = row.rowIndex;
-
-    const containerHeight = this.container[0][0].offsetHeight;
-    const metadataHeight =
-      this.totalMetadataHeight + this.totalRowAddLinkHeight;
-    const rowOffset =
-      containerHeight / 4 - (this.cellYPosition(row) + metadataHeight);
-    this.pan(0, rowOffset, true);
-
-    // Briefly highlight the focused row.
-    for (let i = 0; i < this.rowLabels.length; i++) {
-      this.rowLabels[i].shaded = i !== rowIndex;
-    }
-    this.updateLabelHighlights(
-      this.gRowLabels.selectAll(`.${cs.rowLabel}`),
-      this.rowLabels
-    );
-    this.updateCellHighlights();
-
-    for (let i = 0; i < this.rowLabels.length; i++) {
-      this.rowLabels[i].shaded = false;
-    }
-    setTimeout(() => {
-      this.rowLabels[rowIndex].highlighted = false;
-      this.updateCellHighlights();
+    if (row) {
```

Comment:
When is row null?

Reply:
A user could potentially select and deselect a taxon quickly, in which case scrollToRow would be called with the label but the heatmap would have no longer have the corresponding row information to access, causing the heatmap to crash.